pkg/kubectl/scale_test.go

```diff
@@ -785,3 +806,273 @@ func TestValidateDeployment(t *testing.T) {
 		}
 	}
 }
+
+// TestGenericScaleSimple exercises GenericScaler.ScaleSimple method
+func TestGenericScaleSimple(t *testing.T) {
+	// test data
+	discoveryResources := []*metav1.APIResourceList{
+		{
+			GroupVersion: appsv1beta2.SchemeGroupVersion.String(),
+			APIResources: []metav1.APIResource{
+				{Name: "deployments", Namespaced: true, Kind: "Deployment"},
+				{Name: "deployments/scale", Namespaced: true, Kind: "Scale", Group: "apps", Version: "v1beta2"},
+			},
+		},
+	}
+	appsV1beta2Scale := &appsv1beta2.Scale{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "Scale",
+			APIVersion: appsv1beta2.SchemeGroupVersion.String(),
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "abc",
+		},
+		Spec: appsv1beta2.ScaleSpec{Replicas: 10},
+		Status: appsv1beta2.ScaleStatus{
+			Replicas: 10,
+		},
+	}
+	pathsResources := map[string]runtime.Object{
+		"/apis/apps/v1beta2/namespaces/default/deployments/abc/scale": appsV1beta2Scale,
+	}
+
+	scaleClient, err := fakeScaleClient(discoveryResources, pathsResources)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// test scenarios
+	scenarios := []struct {
+		precondition ScalePrecondition
+		newSize      int
+		targetGR     schema.GroupResource
+		resName      string
+		scaleGetter  scale.ScalesGetter
+		expectError  bool
+	}{
+		// scenario 1: scale up the "abc" deployment
+		{
+			precondition: ScalePrecondition{10, ""},
+			newSize:      20,
+			targetGR:     schema.GroupResource{"apps", "deployment"},
+			resName:      "abc",
+			scaleGetter:  scaleClient,
+		},
+		// scenario 2: scale down the "abc" deployment
+		{
+			precondition: ScalePrecondition{20, ""},
+			newSize:      5,
+			targetGR:     schema.GroupResource{"apps", "deployment"},
+			resName:      "abc",
+			scaleGetter:  scaleClient,
+		},
+		// scenario 3: precondition error, expected size is 1,
+		// note that the previous scenario (2) set the size to 5
+		{
+			precondition: ScalePrecondition{1, ""},
+			newSize:      5,
+			targetGR:     schema.GroupResource{"apps", "deployment"},
+			resName:      "abc",
+			scaleGetter:  scaleClient,
+			expectError:  true,
+		},
+		// scenario 4: precondition is not validated when the precondition size is set to -1
+		{
+			precondition: ScalePrecondition{-1, ""},
+			newSize:      5,
+			targetGR:     schema.GroupResource{"apps", "deployment"},
+			resName:      "abc",
+			scaleGetter:  scaleClient,
+		},
+		// scenario 5: precondition error, resource version mismatch
+		{
+			precondition: ScalePrecondition{5, "v1"},
+			newSize:      5,
+			targetGR:     schema.GroupResource{"apps", "deployment"},
+			resName:      "abc",
+			scaleGetter:  scaleClient,
+			expectError:  true,
+		},
+	}
+
+	// act
+	for index, scenario := range scenarios {
+		t.Run(fmt.Sprintf("scenario %d", index+1), func(t *testing.T) {
```

Comment:
putting a short, descriptive name in the scenario struct will help us remember what these are all doing and give a better name here.

```go
[] struct{
    name string
    .....
```